wavesexchange_topic/src/lib.rs

```diff
@@ -303,10 +321,88 @@ mod parse_and_format {
                             return Err(TopicParseError::InvalidLeasingBalanceTopic);
                         }
                     }
+                    TopicKind::Pairs => {
+                        Topic::extract_exchange_pairs_from_query(&url)?;
```

Comment:
Proper canonicalization might be needed here (see above how the State is implemented)